src/gui/settingswidget.cpp

```diff
@@ -79,17 +78,36 @@ SettingsWidget::SettingsWidget(QWidget *parent) :
     // Read auto update settings
     ui->autoFWBox->setChecked(!settings.value("DisableAutoFWCheck").toBool());
 
-    // Read auto run settings
-    if(!AutoRun::available())
-        ui->loginItemBox->hide();
-    else {
-        if(!AutoRun::once())
-            // If this is the first time running the app, enable auto run by default
-            AutoRun::enable();
-        ui->loginItemBox->setChecked(AutoRun::isEnabled());
+#ifdef Q_OS_LINUX
+    static const QString deletedAutorunPath = QStringLiteral("Program/DeletedOldAutostart");
+    // If autostart was enabled, and there's an autostart file in XDG_CONFIG_DIRS, then delete the local one.
+    if(!CkbSettings::get(deletedAutorunPath).toBool()) {
+        CkbSettings::set(deletedAutorunPath, QVariant::fromValue<bool>(true));
+
+        // Have to do this manually since Qt won't separate system paths from user ones
+        QStringList config_dirs = QString::fromLocal8Bit(qgetenv("XDG_CONFIG_DIRS")).split(QChar(':'), Qt::SkipEmptyParts);
+        if(config_dirs.isEmpty())
+            config_dirs.append(QStringLiteral("/etc/xdg"));
+
+        for(const QString& str : config_dirs) {
+            QDir d(str);
+            if(!d.exists() || !d.cd(QStringLiteral("autostart")))
+                continue;
+
+            if(d.exists(QStringLiteral("ckb-next.desktop"))) {
+                qDebug() << "Found autostart file in" << d.absolutePath();
+                // Path copy pasted from the old autorun.cpp
+                static const QDir path(QStandardPaths::writableLocation(QStandardPaths::HomeLocation) + "/.config/autostart");
+                // ckb-next wouldn't let you modify the autostart file before, so if the file exists, it's a pretty good guarantee that it's the one we created
+                // If the user modified it and made it read only, then this won't be able to remove it anyway
+                QFile::remove(path.absoluteFilePath(QStringLiteral("ckb-next.autostart.desktop")));
```

Comment:
I wonder how important it is, that we inspect at runtime and every start of the GUI, whether an autostart file exists? Especially if the idea is to hoist the responsibility for autostarting to the user anyway.

On the other hand I am using a Desktop Environment, that does not automatically autostart desktop files in a location. So having those left over would not make a difference.